src/IdentityDecorator.php

```diff
@@ -0,0 +1,135 @@
+<?php
+/**
+ * CakePHP(tm) : Rapid Development Framework (http://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * @link          http://cakephp.org CakePHP(tm) Project
+ * @license       http://www.opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Authorization;
+
+use ArrayAccess;
+use InvalidArgumentException;
+
+/**
+ * An decorator implementing the IdentityInterface.
+ *
+ * This decorator is intended to wrap the application defined identity
+ * object and proxy attributes/methods to and 'mixin' the can() method.
+ *
+ * The decorated identity must implement ArrayAccess or be an array.
+ */
+class IdentityDecorator implements IdentityInterface
+{
+    /**
+     * Identity data
+     *
+     * @var array|\ArrayAccess
+     */
+    protected $identity;
+
+    /**
+     * Authorization Service
+     *
+     * @var \Authorization\AuthorizationServiceInterface
+     */
+    protected $authorization;
+
+    /**
+     * Constructor
+     *
+     * @param \Authorization\AuthorizationServiceInterface $service The authorization service.
+     * @param array|\ArrayAccess $identity Identity data
+     * @throws InvalidArgumentException When invalid identity data is passed.
+     */
+    public function __construct(AuthorizationServiceInterface $service, $identity)
+    {
+        if (!is_array($identity) && !$identity instanceof ArrayAccess) {
+            $type = is_object($identity) ? get_class($identity) : gettype($identity);
+            $message = sprintf('Identity data must be an `array` or implement `ArrayAccess` interface, `%s` given.', $type);
+            throw new InvalidArgumentException($message);
+        }
+
+        $this->authorization = $service;
+        $this->identity = $identity;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function can($action, $resource)
+    {
+        return $this->authorization->can($this, $action, $resource);
+    }
+
+    /**
+     * Delegate unknown methods to decorated identity.
+     *
+     * @param string $method The method being invoked.
+     * @param arry $args The arguments for the method.
+     * @return mixed
+     */
+    public function __call($method, $args)
+    {
+        $call = [$this->identity, $method];
```

Comment:
Needs a `is_object` check probably, as the identity may be an array.

Reply:
done.